python/cuml/test/test_qn.py

```diff
@@ -0,0 +1,459 @@
+                                                            [0.8060587]]),
+                                                   decimal=3)
+                elif l2_strength == 0.0:
+                    assert (qn.objective - 0.44295936822891235) < tol
+                    np.testing.assert_almost_equal(qn.coef_.copy_to_host(),
+                                                   np.array([[-1.6899368],
+                                                            [1.9021575],
+                                                            [0.8057671]]),
+                                                   decimal=3)
+                else:
+                    assert (qn.objective - 0.467987984418869) < tol
+                    np.testing.assert_almost_equal(qn.coef_.copy_to_host(),
+                                                   np.array([[-1.3727235],
+                                                            [1.4639963],
+                                                            [0.79312485]]),
+                                                   decimal=3)
+            else:
+                if l1_strength == 0.0 and l2_strength == 0.0:
+                    assert (qn.objective - 0.4317452311515808) < tol
+                    np.testing.assert_almost_equal(qn.coef_.copy_to_host(),
+                                                   np.array([[-2.120777],
+                                                            [3.056865]]),
+                                                   decimal=3)
+                elif l1_strength == 0.0:
+                    assert (qn.objective - 0.4750209450721741) < tol
+                    np.testing.assert_almost_equal(qn.coef_.copy_to_host(),
+                                                   np.array([[-1.3931049],
+                                                            [2.0140104]]),
+                                                   decimal=3)
+
+                elif l2_strength == 0.0:
+                    assert (qn.objective - 0.4769895672798157) < tol
+                    np.testing.assert_almost_equal(qn.coef_.copy_to_host(),
+                                                   np.array([[-1.6214856],
+                                                            [2.3650239]]),
+                                                   decimal=3)
+                else:
+                    assert (qn.objective - 0.5067970156669617) < tol
+                    np.testing.assert_almost_equal(qn.coef_.copy_to_host(),
+                                                   np.array([[-1.2102532],
+                                                            [1.752459]]),
+                                                   decimal=3)
+
+                print()
+
+    elif penalty == 'softmax':
+        pytest.skip("Better initial conditions for softmax tests are "
+                    "in progress.")
+
+    # todo add tests for softmax dtype=np.float64
```

Comment:
Add an issue for this if we aren't doing it in this PR

Reply:
Absolutely, just debugging CI and will open if I'm not mistaken 2 issues regarding tests